src/NUnitFramework/framework/Internal/Commands/SetUpTearDownItem.cs

```diff
@@ -50,7 +50,51 @@ public void RunSetUp(TestExecutionContext context)
             _setUpWasRun = true;
 
             foreach (IMethodInfo setUpMethod in _setUpMethods)
-                RunSetUpOrTearDownMethod(context, setUpMethod);
+            {
+                try
+                {
+                    CallBeforeSetUpsHook(context, setUpMethod);
+                    RunSetUpOrTearDownMethod(context, setUpMethod);
+                }
+                finally
+                {
+                    CallAfterSetUpsHook(context, setUpMethod);
+                }
+            }
+        }
+
+        private void CallBeforeSetUpsHook(TestExecutionContext context, IMethodInfo setUpMethod)
```

Comment:
These methods look identical, except for the two methods to be called.  
Suggest you send those two methods in as parameters to a single method,  to avoid having to check identical code later.  (DRY principle)

Reply:
This pull request I have closed because I missed to design the interface really minimal...
See [this](https://github.com/nunit/nunit/pull/4745) pull request that also does not have the additional methods but only the IHookExtension interface.
Sorry for confusion. Did not found a way to delete the pull request...